packages/pigeon/lib/java_generator.dart

```diff
@@ -158,16 +167,21 @@ static MessageCodec<Object> getCodec() {
               indent.scoped('{', '}', () {
                 final List<String> methodArgument = <String>[];
                 if (method.arguments.isNotEmpty) {
-                  final String argType =
-                      _javaTypeForDartType(method.arguments[0].type);
-                  indent.writeln('@SuppressWarnings("ConstantConditions")');
-                  indent.writeln('$argType input = ($argType)message;');
-                  indent.write('if (input == null) ');
-                  indent.scoped('{', '}', () {
+                  indent.writeln(
+                      'ArrayList<Object> args = (ArrayList<Object>)message;');
+                  enumerate(method.arguments, (int index, NamedType arg) {
+                    final String argType = _javaTypeForDartType(arg.type);
+                    final String argName =
+                        arg.name.isEmpty ? 'arg$index' : '${arg.name}Arg';
```

Comment:
How about declaring this as a function with a comment explaining it, as you did in the Dart code? I think it would non-obvious to a reader who didn't just have the discussions we've just had about collisions why this isn't just calling `_getArgumentName`, whereas the commented function makes it very clear.

Reply:
Done.